src/main/java/br/com/openenade/api/nota/NotaService.java

```diff
@@ -0,0 +1,96 @@
+package br.com.openenade.api.nota;
+
+import java.util.List;
+import java.util.Optional;
+import javax.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import br.com.openenade.api.ano.Ano;
+import br.com.openenade.api.ano.AnoRepository;
+import br.com.openenade.api.curso.Curso;
+import br.com.openenade.api.curso.CursoId;
+import br.com.openenade.api.curso.CursoRepository;
+import br.com.openenade.api.universidade.Universidade;
+import br.com.openenade.api.universidade.UniversidadeId;
+import br.com.openenade.api.universidade.UniversidadeRepository;
+
+@Service
+@Transactional
+public class NotaService {
+
+    @Autowired
+    private NotaRepository notaRepository;
+
+    @Autowired
+    private AnoRepository anoRepository;
+
+    @Autowired
+    private CursoRepository cursoRepository;
+
+    @Autowired
+    private UniversidadeRepository universidadeRepository;
+
+    public Nota save(Nota nota) {
+
+        this.anoRepository.save(nota.getInfo().getAno());
```

Comment:
I think that is not good to overwrite Curso and Universidade in the database applying consecutive saves for each Nota insertion.

Reply:
It was only to short, we will fix it in the future.

Reply:
@Icaro-Lima this issue is not resolved.